### ISSUE TITLE
Connect exception when Chrome process is started by cdp4j

### ISSUE THREAD

**Issue by gschulze** [webfolderio/cdp4j#95]
I'm hitting the following exception when attempting to create a browser context if the chrome process is started by cdp4j:

```
Exception in thread "main" io.webfolder.cdp.exception.CdpException: Unable to connect [localhost:9222]
	at io.webfolder.cdp.session.SessionFactory.getVersion(SessionFactory.java:412)
	at io.webfolder.cdp.session.SessionFactory.getBrowserSession(SessionFactory.java:278)
	at io.webfolder.cdp.session.SessionFactory.createBrowserContext(SessionFactory.java:427)
	...
```

I've been able to work around the issue by putting `Thread.sleep(1000)` before `sessionFactory.createBrowserContext()`, however, the code should wait until the remote debugging port ist available before returning from Launcher.launch().

**Comment by gschulze** [comment]
I used the following code to wait for the port to be available:

```
boolean done = false;
while (!done) {
	try(Socket socket = new Socket()) {
		socket.connect(new InetSocketAddress(host, port));
		done = true;
	} catch (IOException e) {
		try {
			Thread.sleep(100);
		} catch(InterruptedException e) {
			// Ignore
		}
	}
}
```

**Comment by pfurbacher** [comment]
I encountered this ```Unable to connect  [localhost:9222]``` problem earlier today as well. Both solutions presented above worked, but it seems to me that there's something lacking in the implementation of ```Launcher.launch(...)``` which could prevent this problem. It shouldn't be necessary for the API consumer to writing such "guard" code to make things work.

Some additional info: I encountered this problem in unit testing my code which uses headless mode to print to PDF. The first time I ran the test, it would fail with this "unable to connect" problem. Then, subsequent tests succeeded, most likely because an old instance was already listening on port 9222. (I was shocked to see so many previous headless chrome instances were just hanging around in "Activity Monitor" -- didn't even consider that you had defaulted to a **NullProcessManager** in ```Launcher```. Really?!) 

Once I manually killed off all those previous instances, the "unable to connect" problem would come back the first time I ran a test. And of course, once I implemented a MacOSProcessManager which killed off the process correctly, the problem was endemic, until I found this issue and its workarounds.

**Comment by webfolderio** [comment]
hi @pfurbacher 

> didn't even consider that you had defaulted to a NullProcessManager in Launcher. Really?!

:) ProcessManager added with version 2.x. We don't want to brake the api, thats why the default is NullProcessManager. I should improve and update the document.

> And of course, once I implemented a MacOSProcessManager

Unfortunately we do not have Mac environment to test cdp4j.
Did you try to use [DefaultProcessManager.java](https://github.com/webfolderio/cdp4j/blob/master/src/main/java/io/webfolder/cdp/DefaultProcessManager.java)? It's platform independent implementation but requires Java 9.

Finally, i am going to improve `Launcher.launch(...)` with version 3.1 which will be released at the beginning of June.

**Comment by webfolderio** [comment]
> contribute to your Java 8 branch (where is it?) 

There is only master branch. I will happy to merge your MacOSProcessManager.

>It shouldn't be necessary for the API consumer to writing such "guard" code to make things work.

`AbstractLauncher.launch()` method improved with version 3.0.1-SNAPSHOT
New implementation checks if Chrome is ready and WebSocket connection is alive.

**Comment by webfolderio** [comment]
[MacOsProcessManager.java](https://github.com/webfolderio/cdp4j/blob/master/src/main/java/io/webfolder/cdp/MacOsProcessManager.java) is ready to use and will be released with version 3.0.3
